crates/webidl/src/lib.rs

```diff
@@ -118,7 +133,234 @@ impl WebidlParse for webidl::ast::NonPartialInterface {
+        // generating any bindings for this function.
+        webidl::ast::TypeKind::ArrayBuffer
+        | webidl::ast::TypeKind::ByteString
+        | webidl::ast::TypeKind::DOMString
+        | webidl::ast::TypeKind::DataView
+        | webidl::ast::TypeKind::Error
+        | webidl::ast::TypeKind::Float32Array
+        | webidl::ast::TypeKind::Float64Array
+        | webidl::ast::TypeKind::FrozenArray(_)
+        | webidl::ast::TypeKind::Int16Array
+        | webidl::ast::TypeKind::Int32Array
+        | webidl::ast::TypeKind::Int8Array
+        | webidl::ast::TypeKind::Object
+        | webidl::ast::TypeKind::Promise(_)
+        | webidl::ast::TypeKind::Record(..)
+        | webidl::ast::TypeKind::Sequence(_)
+        | webidl::ast::TypeKind::Symbol
+        | webidl::ast::TypeKind::USVString
+        | webidl::ast::TypeKind::Uint16Array
+        | webidl::ast::TypeKind::Uint32Array
+        | webidl::ast::TypeKind::Uint8Array
+        | webidl::ast::TypeKind::Uint8ClampedArray
+        | webidl::ast::TypeKind::Union(_) => {
+            return None;
+        }
+    })
+}
+
+fn simple_fn_arg(ident: proc_macro2::Ident, ty: syn::Type) -> syn::FnArg {
+    syn::FnArg::Captured(syn::ArgCaptured {
+        pat: syn::Pat::Ident(syn::PatIdent {
+            by_ref: None,
+            mutability: None,
+            ident,
+            subpat: None,
+        }),
+        colon_token: Default::default(),
+        ty,
+    })
+}
+
+impl<'a> WebidlParse<'a> for webidl::ast::RegularOperation {
+    type Extra = &'a str;
+
+    fn webidl_parse(&self, program: &mut backend::ast::Program, self_name: &'a str) -> Result<()> {
+        let fn_name = match self.name {
+            None => return Ok(()),
+            Some(ref name) => Ident::new(name, proc_macro2::Span::call_site()),
+        };
+
+        let (output, ret) = match self.return_type {
+            webidl::ast::ReturnType::Void => (syn::ReturnType::Default, None),
+            webidl::ast::ReturnType::NonVoid(ref ty) => match webidl_ty_to_syn_ty(ty) {
+                None => return Ok(()),
+                Some(ty) => (
+                    syn::ReturnType::Type(Default::default(), Box::new(ty.clone())),
+                    Some(ty),
+                ),
+            },
+        };
+
+        let mut inputs = Vec::with_capacity(self.arguments.len() + 1);
+        let mut arguments = Vec::with_capacity(self.arguments.len() + 1);
+
+        let self_ty = simple_path_ty(Some(self_name));
+        let self_ref_ty = shared_ref(self_ty.clone());
+        inputs.push(simple_fn_arg(
+            proc_macro2::Ident::new("self_", proc_macro2::Span::call_site()),
+            self_ref_ty.clone(),
+        ));
+        arguments.push(self_ref_ty);
+
+        for arg in &self.arguments {
+            if arg.optional || arg.variadic {
+                // We don't support optional or variadic functions yet; skip
+                // bindings for this this whole function.
```

Comment:
I wonder if for debugging purposes at some point if it'd be good to log whenever a function is skipped with a brief reason as to why?

Reply:
Yeah, I was wondering about this too. Do you have a preferred method of logging? `env_logger`?

Reply:
Sure yeah, that should work!

Reply:
Will add logging in a follow up PR.